src/components/fx/hover.js

```diff
@@ -1095,7 +1095,9 @@ function cleanPoint(d, hovermode) {
         d.yVal = d.ya.c2d(d.yLabelVal);
     }
 
-    if(d.zLabelVal !== undefined) d.zLabel = String(d.zLabelVal);
+    if(d.zLabelVal !== undefined && d.zLabel === undefined) { // Traces like heatmaps generate the zLabel in their hoverPoints function
```

Comment:
Hmm. Too bad we need this. Good catch :baseball:  `cleanPoint` could use a little bit of a refactor. 

Would you mind putting that `//  ` comment about the `if()` statement?